codegen/manual_client.cpp

```diff
@@ -792,3 +799,44 @@ extern "C"
         }
     }
 }
+
+cudaError_t cudaFree(void* devPtr)
+{
+    cudaError_t return_value;
+    void *maybe_ptr = maybe_free_unified_mem(0, devPtr);
+
+    if (maybe_ptr != nullptr) {
+        if (rpc_start_request(0, RPC_cudaFree) < 0 ||
+            rpc_write(0, &maybe_ptr, sizeof(void*)) < 0 ||
+            rpc_wait_for_response(0) < 0 ||
+            rpc_end_response(0, &return_value) < 0)
+            return cudaErrorDevicesUnavailable;
+    } else {
+        if (rpc_start_request(0, RPC_cudaFree) < 0 ||
+            rpc_write(0, &devPtr, sizeof(void*)) < 0 ||
+            rpc_wait_for_response(0) < 0 ||
+            rpc_end_response(0, &return_value) < 0)
+            return cudaErrorDevicesUnavailable;
+    }
```

Comment:
Can this be one block instead?
```suggestion
    maybe_free_unified_mem(0, devPtr);

        if (rpc_start_request(0, RPC_cudaFree) < 0 ||
            rpc_write(0, &devPtr, sizeof(void*)) < 0 ||
            rpc_wait_for_response(0) < 0 ||
            rpc_end_response(0, &return_value) < 0)
            return cudaErrorDevicesUnavailable;
```